salvador/APISecret.swift

```diff
@@ -0,0 +1,10 @@
+//
+//  apiSecret.swift
+//  salvador
+//
+//  Created by Geury Roustand on 07.02.23.
+//
+
+import Foundation
+
+var APISecret = "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"
```

Comment:
Just can't emphasis  this enough: don't check in your API key ;)

Reply:
I don't know why the .gitIgnore is still pushing the API key even though I added the file APISicret, do you know if I'm missing something here?


![Screenshot 2023-02-13 at 09 16 52](https://user-images.githubusercontent.com/61476699/218405522-467a5714-650f-4d69-aa31-ab170a01ebd0.png)

Reply:
I don't see the gitignore file. At least not in this PR. You need to add and commit it to the repo.

However, that still might not be a good solution to ignore a swift file because the code won't compile for someone that checks out the repo. I would rather put the key in a JSON file (as key-value pair), which you do not check in and put to the gitignore.

Then you add some code to parse the API key from the JSON file in runtime. If parsing fails you can display a proper error message to the user.

Reply:
Normally all the files inside the .gitignore should be ignored regardless of file type format,  I can create a JSON file but I'm not sure if will work.

Reply:
The file type doesn't matter, that's true. gitignore will ignore it. My point is, that if you ignore a swift file that is part of the app, nobody that checks out your app will be able to build it out of the box. When you use the approach I explained, it won't be the case. Maybe the JSON file with the key is not there and the API request will fail. But your app will compile in any case.

Reply:
Sure, there is some extra effort to make everything work by adding the JSON file with a valid key. But this is something that needs to be only set up once e.g. explained in a `readme`.  It is more elegant than ignoring parts of the compilable code.

Anyway, it is not a must to do that, if you are fine with the current solution. I only make suggestions. It's on you to apply it or not depending on if you think it is a good idea or not. Everything is definitely better than having your plain API key checked in.